Setting rwsetV1ProtoBytes as binary using local .gitattributes

cendhu opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->

## Description
<!-- Describe your changes in detail. -->

## Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Fixes #

## How Has This Been Tested?
<!-- If this PR does not contain a new test case, explain why. -->
<!-- Describe in detail how you tested your changes. -->

## Checklist:
<!-- To check a box, and an 'x': [x] -->
<!-- To uncheck box, add a space: [ ] -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [] I have either added documentation to cover my changes or this change requires no new documentation.
- [] I have either added unit tests to cover my changes or this change requires no new tests.
- [] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

<!-- The continuous integration build process will run [make checks](https://github.com/hyperledger/fabric/blob/master/Makefile#L22) to confirm that tests pass and that code quality meets minimum standards. You may optionally run this locally as PRs will not be accepted until they pass. -->

Signed-off-by:

manish-sethi committed:
interfaces for private read-write permanent store and transient store

Change-Id: Ieabf014d64d968efdd320e414ff4db5099d8ab3b
Signed-off-by: manish <manish.sethi@gmail.com>

manish-sethi committed:
Added new interfaces for supporting private data handling in ledger

Change-Id: Ife846b5fbca814bdc1831e6f750fb073e2e51616
Signed-off-by: manish <manish.sethi@gmail.com>

manish-sethi committed:
Changed 'Temp' to 'Transient'

Change-Id: I03455748d1eabccdb4b2a670a499fba9b93053ab
Signed-off-by: manish <manish.sethi@gmail.com>

cendhu committed:
Merge pull request #2 from manish-sethi/manish-sidedb-dev

interfaces for private read-write permanent store and transient store

cendhu committed:
Add hashed and private read-write set to proto

This CR adds protobuf struct for hashed read/write set and private
read/write set.

To the existing KVRWSet, i have added a field to store hashed read write
set of all collections. For a private read, we can store the hash in key
field but keep the version as is. For private writes, we can store hash
of key and hash of value but store the IsDelete as is. Hence, reused
existing KVRead and KVWrite.

For the range query info, it does not make sense to store the hash of
each field in the RangeQueryInfo struct. Note that we cannot do a range
query on state DB using hashed startKey and endKey to find phantom
reads.

Instead of hashed RangeQueryInfo per collection, i have added a string
field to hold a hash of entire private RangeQueryInfos of that collection.
During validation, we can compare the hash of entire private RangeQueryInfo
present in the tempDB with the hash present in the block. I thought of
doing the same for hashed write-set too as we may not store this value in stateDB.
However, it might be useful during validation phase (while comparing the
write set of Ti with read set of Ti+1 -- especially when the private
write-set is unavailable/not received).

If the peer is configured to move on without private write-set
(implicit: phantom read check is disabled), we must not use
RangeQueryInfo. Instead add the read <key, version> to private read
set and hashed read set.

I have also added PrivateCollectionsKVRWSet to hold private read-write
set of all collections. For clear differentiation on variable names, i
have used Reads, HashedReads, and PrivateReads (similarly for the Writes).

manish-sethi commented:
Senthil, I think that you pushed your changes from your existing branch to this new branch. It shows me as if you want to merge 9 commits (which are from the previous branch). I guess, that you would need to get a new copy of this branch in your local and then shift your changes from your  existing local to new local before pushing a PR.

cendhu committed:
Setting rwsetV1ProtoBytes as binary

Instead of modifying the .gitarrtibutes in the root directory of fabric,
we add a new attributes files in the directory where rwsetV1ProtoBytes
is located.
